core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1054,7 +1059,6 @@ class KafkaController(val config: KafkaConfig,
   }
 
   private def fetchTopicDeletionsInProgress(): (Set[String], Set[String]) = {
-    val controllerContextSnapshot = ControllerContextSnapshot(controllerContext)
```

Comment:
I'd actually meant to ask Hao or Joseph (or whoever) about that:  is there a locking or consistency concern?  If so, the correct fix would be to use this val two lines down.  (I wasn't able to convince myself either way, however.)

Reply:
The ControllerContextSnapshot was introduced to speed up the ControllerContext.isReplicaOnline method https://github.com/linkedin/kafka/pull/206/files
In the original PR, I think there is no consistency concern since the ControllerContextSnapshot object was used in a read-only manner in each individual caller.
In places where the isReplicaOnline is not called, there is no need to instantiate a ControllerContextSnapshot.

In 3.0, the fetchTopicDeletionsInProgress no longer needs to call isReplicaOnline due to the change of behavior in friction-less topic deletion. Thus it's safe to drop the previously introduced controllerContextSnapshot variable.